lib/util/newDefaultObject.js

```diff
@@ -0,0 +1,31 @@
+import isEmpty from './isEmpty';
+
+function isArrayIndex(n) {
+  return /^[0-9]+$/.test(n);
```

Comment:
let's use this one: 

```js
function isInt(value) {
  if (isNaN(value)) {
    return false;
  }
  var x = parseFloat(value);
  return (x | 0) === x;
}
```

via http://stackoverflow.com/questions/14636536/how-to-check-if-a-variable-is-an-integer-in-javascript